File: 132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        stack = []
        x = -((10**9)+1)
        for i in range(len(nums)-1,-1,-1):
            if nums[i] < x:
                return True
            while stack and stack[-1] < nums[i]:
                x = stack.pop()
            stack.append(nums[i])
        return False

File: A2SV_programming/132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        stack<int>st;
        long x = -1e9;
        for (int i = nums.size()-1;i >= 0;i--){
            if (nums[i] < x){
                return true;
            }
            while (st.size() > 0 && st.top() < nums[i]){
                x = st.top();
                st.pop();
            }
            st.push(nums[i]);
        }
        return false;

    }
};